app/models/grda_warehouse/hud/client.rb

```diff
@@ -1895,19 +1895,12 @@ def self.text_search(text, client_scope:)
         where = sa[:PersonalID].eq(text).or(sa[:id].eq(text))
       else
         query = "%#{text}%"
-        alt_names = UniqueName.where(double_metaphone: Text::Metaphone.double_metaphone(text).to_s).map(&:name)
-        nicks = Nickname.for(text).map(&:name)
         where = sa[:FirstName].matches(query).
           or(sa[:LastName].matches(query))
-        if nicks.any?
-          nicks_for_search = nicks.map { |m| GrdaWarehouse::Hud::Client.connection.quote(m) }.join(',')
-          where = where.or(nf('LOWER', [arel_table[:FirstName]]).in(nicks_for_search))
-        end
-        if alt_names.present?
-          alt_names_for_search = alt_names.map { |m| GrdaWarehouse::Hud::Client.connection.quote(m) }.join(',')
-          where = where.or(nf('LOWER', [arel_table[:FirstName]]).in(alt_names_for_search)).
-            or(nf('LOWER', [arel_table[:LastName]]).in(alt_names_for_search))
-        end
+
+        where = nickname_search(where, text)
+        where = metaphone_search_first_name(where, text)
+        where = metaphone_search_last_name(where, text)
       end
       begin
```

Comment:
I am trying out the graphql query and I can't get any results when using the omni-search.

I narrowed it down to the lines below, where we do `map { |m| m.destination_client&.id }.`. For me, if I replace that with `pluck(:id)`, I get results. I think that's because those destination clients are not linked up to my hmis warehouse data source.

Should we be looking at source client IDs here instead of destination client IDs, for the HMIS?